packages/react-components/react-avatar/src/components/AvatarGroup/AvatarGroup.types.ts

```diff
@@ -10,63 +10,64 @@ export type AvatarGroupSlots = {
   /**
    * Popover trigger slot that can be used to change the overflow indicator.
    */
-  popoverTrigger?: Slot<typeof Button>;
+  popoverTrigger: NonNullable<Slot<typeof Button>>;
 
   /**
    * Popover surface that will be displayed when the popover is triggered.
    */
-  popoverSurface?: Slot<typeof PopoverSurface>;
+  popoverSurface: NonNullable<Slot<typeof PopoverSurface>>;
 };
 
-type AvatarGroupCommons = {
+/**
+ * AvatarGroup Props
+ */
+export type AvatarGroupProps = ComponentProps<Partial<AvatarGroupSlots>> & {
   /**
    * Layout the Avatars should be displayed as.
    * @default spread
    */
-  layout: 'spread' | 'stack' | 'pie';
+  layout?: 'spread' | 'stack' | 'pie';
 
   /**
    * Maximum number of Avatars to be displayed before overflowing.
    * NOTE: if pie layout is used, `maxAvatars` will be ignored.
    * @default 5
    */
-  maxAvatars: number;
-
-  /**
-   * Size of the avatars.
-   * @default 32
-   */
-  size: AvatarSizes;
+  maxAvatars?: number;
 
   /**
    * Whether the overflow indicator should render an icon instead of the number of overflowed avatars.
    * @default false
    */
-  overflowIndicator: 'number-overflowed' | 'icon';
+  overflowIndicator?: 'number-overflowed' | 'icon';
 
   /**
-   * Strings for localizing text in the tooltip.
+   * Size of the avatars.
+   * @default 32
    */
-  strings?: AvatarGroupStrings;
+  size?: AvatarSizes;
 };
 
-/**
- * AvatarGroup Props
- */
-export type AvatarGroupProps = ComponentProps<AvatarGroupSlots> & Partial<AvatarGroupCommons>;
-
 /**
  * State used in rendering AvatarGroup
  */
 export type AvatarGroupState = ComponentState<AvatarGroupSlots> &
-  AvatarGroupCommons & {
+  Required<Pick<AvatarGroupProps, 'layout' | 'maxAvatars' | 'size' | 'overflowIndicator'>> & {
+    /**
+     * Tooltip content for the overflow indicator.
+     */
     tooltipContent: TooltipProps['content'];
+
+    /**
+     * Whether there are more Avatars than `maxAvatars`.
+     * @default false
+     */
+    hasOverflow: boolean;
```

Comment:
_nit:_ Order these alphabetically

```suggestion
    /**
     * Whether there are more Avatars than `maxAvatars`.
     * @default false
     */
    hasOverflow: boolean;
    
    /**
     * Tooltip content for the overflow indicator.
     */
    tooltipContent: TooltipProps['content'];
```